### PR TITLE
40-0247 - Build front-end

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Build Presidential Memorial Certificate (40-0247) front-end.  NOTE: Refactored shared IntroductionPageView.jsx component to re-surface `<va-omb-info/>` component's support for custom Privacy-Act-Statement modal-content.
- Team: Platform VA Product Forms
- \[Flipper not implemented yet\]

## Related issue(s)

- _Link to ticket created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#453

## Testing done

- Manual UI testing was successful.  Automated tests to follow, via separate ticket.

## What areas of the site does it impact?

This form ONLY.

## Requested Feedback

In addition to the regular code-review:

- @ndsprinkle / @rhasselle-oddball / @Thrillberg , do we need a transformForSubmit config-prop when there's no data-transformation necessary?  I did add one that has no transformations right now, but I wasn't sure if I could omit the prop itself.  Please let me know.  Thanks much!

### CODE REVIEW

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import {
+  schema as addressSchema,
+  uiSchema as addressUiSchema,
+} from 'platform/forms/definitions/address.js';
+import definitions from 'vets-json-schema/dist/definitions.json';
+
+export default {
+  uiSchema: {
+    additionalAddress: addressUiSchema(
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:11]: Can you use the web component version here? addressUI() / addressSchema.  And for the title "Tell us where you'd like us..." can be a separate 'view:something'.

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          </dt>
+          <dd>{children}</dd>
+        </div>
+      ),
+    },
+  },
+  schema: {
+    type: 'object',
+    properties: {
+      additionalAddress: addressSchema({ definitions }, true),
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:48]: same here with the web component version

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesYesNo.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// can't use v3 yesNoUI because it doesn't support JSX ui:title
+/* @type {PageSchema} */
+export default {
+  uiSchema: {
+    additionalCertificates: {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/additionalCertificatesYesNo.js:5]: can you use the web component yes no here? you can see yesNoPattern for example. either you can use yesNoUI/yesNoSchema directly, or include the `ui:webComponentField: YesNoField`

File: src/applications/simple-forms/40-0247/pages/requestType.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+const labelText =
+  'Is this your first time requesting a Presidential Memorial Certificate?';
+
+/* @type {PageSchema} */
+export default {
+  uiSchema: {
+    isFirstRequest: {
+      'ui:title': labelText, // yesNo widget doesn't support JSX
+      'ui:widget': 'yesNo',
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/requestType.js:11]: same

File: src/applications/simple-forms/40-0247/pages/veteranIdInfo.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import definitions from 'vets-json-schema/dist/definitions.json';
+import ssnUI from 'platform/forms-system/src/js/definitions/ssn';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    veteranSSN: ssnUI,
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/veteranIdInfo.js:7]: can you use a web component pattern(s) here?  there is ssnUI/Schema, vaFileNumberUI/Schema, or ssnOrVaFileNumberUI/Schema.

In the ssnOrVaFileNumber version, only one is required, which can be used like
const ssnUI = ssnOrVaFileNumberUI()
const ssnSchema = ssnOrVaFileNumberSchema

veteranSSN: ssnUI.ssn,
veteranVaFileNumber: ssnUI.vaFileNumber

File: src/applications/simple-forms/40-0247/pages/veteranPersonalInfo.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
+    'ui:title': (
+      <h3 className="vads-u-margin-y--0">
+        Tell us who you’re requesting a Presidential Memorial Certificate for
+      </h3>
+    ),
+    veteranFullName: fullNameNoSuffixUI(),
+    veteranDateOfBirth: currentOrPastDateUI('Date of birth'),
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/40-0247/pages/veteranPersonalInfo.js:20]: we also have dateOfBirth and dateOfDeath patterns explicitly.  In those, I think the error message is different.

**Review by rhasselle-oddball — COMMENTED** [review]
changes look good. can you swap out for web component versions for the following cases?

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js (hunk 1 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:11]: Refactored to use v3 addressNoMilitaryUI & addressNoMilitarySchema.  Turns out we actually _can_ mix v3-web-components with older patterns/uiSchemas/schemas.

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js (hunk 2 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:48]: Refactored to use v3 addressNoMilitarySchema.

File: src/applications/simple-forms/40-0247/pages/additionalCertificatesYesNo.js (hunk 3 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesYesNo.js:5]: Refactored to use v3 yesNoUI & yesNoSchema, but how do we customize the error-messages?

Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/additionalCertificatesYesNo.js:5]: Refactored to use v3 yesNoUI & yesNoSchema, but can we customize error-message?

File: src/applications/simple-forms/40-0247/pages/requestType.js (hunk 4 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/requestType.js:11]: Refactored to use v3 yesNoUI & yesNoSchema, but couldn't customize error-message.

File: src/applications/simple-forms/40-0247/pages/veteranIdInfo.js (hunk 5 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/veteranIdInfo.js:7]: Refactored to use v3 ssnOrVaFileNumberUI & ssnOrVaFileNumberUI.

File: src/applications/simple-forms/40-0247/pages/veteranPersonalInfo.js (hunk 6 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/veteranPersonalInfo.js:20]: Refactored to use dateOfBirth & dateOfDeath patterns.

File: src/applications/simple-forms/40-0247/tests/fixtures/data/test-data.json
Comment (tlei123) [inline src/applications/simple-forms/40-0247/tests/fixtures/data/test-data.json:1]: This was moved into the correct subdirectory.

File: src/applications/simple-forms/shared/components/IntroductionPageView.jsx [hunk 7]
```diff
@@ -47,11 +47,21 @@ export const IntroductionPageView = ({
       </SaveInProgressIntro>
       {additionalChildContent || null}
       <p />
-      <va-omb-info
-        res-burden={resBurden}
-        omb-number={ombNumber}
-        exp-date={expDate}
-      />
+      {!customPrivacyActStmt ? (
```
Comment (tlei123) [inline src/applications/simple-forms/shared/components/IntroductionPageView.jsx:50]: This re-surfaces va-omb-info's support for custom Privacy-Act-Statement modal-content.

**Review by rhasselle-oddball — APPROVED** [review]
thanks for all changes. think we discussed the custom label h3 enough and it does seem hard to "fix" so leaving as is seems reasonable while we figure out the right solution on a followup.

**Pull request merged**